src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -1096,6 +1096,87 @@ private static void Parse(
             database.CompleteAllocations();
         }
 
+        private static void ValidateDocument(JsonDocument document, bool allowDuplicateProperties)
```

Comment:
Nit: this seems like a bit of a nondescript name given that its only purpose is to check for duplicates. Perhaps rename this to `ValidateNoDuplicateProperties` without the boolean and change the workhorse method to `ValidateNoDuplicatePropertiesCore`?